pkg/scalers/cron_scaler.go

```diff
@@ -120,48 +108,43 @@ func parseCronTimeFormat(s string) string {
-	var specReplicas int64 = 1
 	externalMetric := &v2.ExternalMetricSource{
 		Metric: v2.MetricIdentifier{
 			Name: GenerateMetricNameWithIndex(s.metadata.TriggerIndex, kedautil.NormalizeString(fmt.Sprintf("cron-%s-%s-%s", s.metadata.Timezone, parseCronTimeFormat(s.metadata.Start), parseCronTimeFormat(s.metadata.End)))),
 		},
-		Target: GetMetricTarget(s.metricType, specReplicas),
+		Target: GetMetricTarget(s.metricType, s.metadata.DesiredReplicas),
 	}
 	metricSpec := v2.MetricSpec{External: externalMetric, Type: cronMetricType}
 	return []v2.MetricSpec{metricSpec}
 }
 
-// GetMetricsAndActivity returns value for a supported metric and an error if there is a problem getting the metric
 func (s *cronScaler) GetMetricsAndActivity(_ context.Context, metricName string) ([]external_metrics.ExternalMetricValue, bool, error) {
-	var defaultDesiredReplicas = int64(defaultDesiredReplicas)
-
 	location, err := time.LoadLocation(s.metadata.Timezone)
 	if err != nil {
-		return []external_metrics.ExternalMetricValue{}, false, fmt.Errorf("unable to load timezone. Error: %w", err)
+		return []external_metrics.ExternalMetricValue{}, false, fmt.Errorf("unable to load timezone: %w", err)
 	}
 
-	// Since we are considering the timestamp here and not the exact time, timezone does matter.
-	currentTime := time.Now().Unix()
+	currentTime := time.Now().In(location)
 
-	nextStartTime, startTimecronErr := getCronTime(location, s.metadata.Start)
-	if startTimecronErr != nil {
-		return []external_metrics.ExternalMetricValue{}, false, fmt.Errorf("error initializing start cron: %w", startTimecronErr)
-	}
+	// Use the pre-parsed schedules to get the next start and end times
+	nextStartTime := getCronTime(location, s.startSchedule)
+	nextEndTime := getCronTime(location, s.endSchedule)
+
+	isWithinInterval := false
 
-	nextEndTime, endTimecronErr := getCronTime(location, s.metadata.End)
-	if endTimecronErr != nil {
-		return []external_metrics.ExternalMetricValue{}, false, fmt.Errorf("error intializing end cron: %w", endTimecronErr)
+	if nextStartTime.Before(nextEndTime) {
+		// Interval within the same day
+		isWithinInterval = currentTime.After(nextStartTime) && currentTime.Before(nextEndTime)
+	} else {
+		// Interval spans midnight
+		isWithinInterval = currentTime.After(nextStartTime) || currentTime.Before(nextEndTime)
 	}
 
-	switch {
-	case nextStartTime < nextEndTime && currentTime < nextStartTime:
-		metric := GenerateMetricInMili(metricName, float64(defaultDesiredReplicas))
-		return []external_metrics.ExternalMetricValue{metric}, false, nil
-	case currentTime <= nextEndTime:
-		metric := GenerateMetricInMili(metricName, float64(s.metadata.DesiredReplicas))
-		return []external_metrics.ExternalMetricValue{metric}, true, nil
-	default:
-		metric := GenerateMetricInMili(metricName, float64(defaultDesiredReplicas))
-		return []external_metrics.ExternalMetricValue{metric}, false, nil
+	metricValue := float64(1)
```

Comment:
This isn't correct because during the not active period, we will return 1 instead of 0
```suggestion
	metricValue := float64(0)
```

Reply:
I found the error. I made a (stupid) mistake in the GetMetricTarget (`Target: GetMetricTarget(s.metricType, specReplicas)`). Here I accidentally passed the `desiredReplicas` instead of de var `specReplicas`, which of course should not be done (and that's why it scales to 1 instead of 4). I adjusted this and tested it on our own cluster. It is all fine now. 

I still see some test errors regarding the Selenium Grid. I think that once #6246 is merged the e2e test can be run again.